cocotb/share/lib/simulator/simulatormodule.c

```diff
@@ -46,6 +46,161 @@ struct module_state {
     PyObject *error;
 };
 
+
+/* define the extension types */
+
+/* for gpi_sim_hdl */
+static PyTypeObject gpi_sim_hdl_Type;
+
+typedef struct {
+    PyObject_HEAD
+    gpi_sim_hdl hdl;
+} gpi_sim_hdl_Object;
+
+static PyObject *gpi_sim_hdl_repr(gpi_sim_hdl_Object *self) {
+    PyTypeObject *type = Py_TYPE(self);
+    return PyUnicode_FromFormat("<%s at %p>",
+                                  type->tp_name, self->hdl);
+}
+static Py_hash_t gpi_sim_hdl_hash(gpi_sim_hdl_Object *self) {
+    Py_hash_t ret = (Py_hash_t)self->hdl;
+    if (ret == (Py_hash_t)-1) {
+        ret = (Py_hash_t)-2;
+    }
```

Comment:
This seems unnecessary, `gpi_sim_hdl`s alignment is not 1, the pointer will never be equal to -1.

Reply:
I wanted to play it safe here. I don't think the standard makes any guarantee about pointer bitpatterns, but python requires that -1 is never returns.